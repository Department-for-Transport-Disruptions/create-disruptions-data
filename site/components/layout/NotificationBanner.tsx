import Link from "next/link";

interface NotificationBannerProps {
    title?: string;
    content: string;
    linkText?: string;
    afterLinkText?: string;
}

const NotificationBanner = ({ title = "Important", content, linkText, afterLinkText }: NotificationBannerProps) => (
    <div
        className="govuk-notification-banner"
        role="region"
        aria-labelledby="govuk-notification-banner-title"
        data-module="govuk-notification-banner"
    >
        <div className="govuk-notification-banner__header">
            <h2 className="govuk-notification-banner__title" id="govuk-notification-banner-title">
                {title}
            </h2>
        </div>
        <div className="govuk-notification-banner__content">
            <p className="govuk-notification-banner__heading">
                {content}
                {linkText && (
                    <Link className="govuk-notification-banner__link" href="#">
                        {linkText}
                    </Link>
                )}
                {afterLinkText ? afterLinkText : ""}
            </p>
        </div>
    </div>
);

export default NotificationBanner;
