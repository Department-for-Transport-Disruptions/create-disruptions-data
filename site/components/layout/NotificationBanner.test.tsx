import renderer from "react-test-renderer";
import { describe, it, expect } from "vitest";
import NotificationBanner from "./NotificationBanner";

describe("NotificationBanner", () => {
    it("should render correctly", () => {
        const tree = renderer.create(<NotificationBanner content={"Banner Text"} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("should render correctly with link", () => {
        const tree = renderer
            .create(
                <NotificationBanner
                    title={"Very Important"}
                    content={"Banner Text"}
                    linkText="www.google.com"
                    afterLinkText="."
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
