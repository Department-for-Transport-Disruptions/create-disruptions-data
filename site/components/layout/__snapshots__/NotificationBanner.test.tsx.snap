// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotificationBanner > should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="govuk-notification-banner-title"
    class="govuk-notification-banner"
    data-module="govuk-notification-banner"
    role="region"
  >
    <div
      class="govuk-notification-banner__header"
    >
      <h2
        class="govuk-notification-banner__title"
        id="govuk-notification-banner-title"
      >
        Important
      </h2>
    </div>
    <div
      class="govuk-notification-banner__content"
    >
      <p
        class="govuk-notification-banner__heading"
      >
        Banner Text
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NotificationBanner > should render correctly with link 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="govuk-notification-banner-title"
    class="govuk-notification-banner"
    data-module="govuk-notification-banner"
    role="region"
  >
    <div
      class="govuk-notification-banner__header"
    >
      <h2
        class="govuk-notification-banner__title"
        id="govuk-notification-banner-title"
      >
        Very Important
      </h2>
    </div>
    <div
      class="govuk-notification-banner__content"
    >
      <p
        class="govuk-notification-banner__heading"
      >
        Banner Text
        <a
          class="govuk-notification-banner__link"
          href="www.google.com"
        >
          look here
        </a>
        .
      </p>
    </div>
  </div>
</DocumentFragment>
`;
